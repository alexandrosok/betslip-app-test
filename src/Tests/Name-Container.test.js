import React from 'react';
import {shallow, configure} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16'
import {shallowToJson} from 'enzyme-to-json';
import NameContainer from "../components/Name-Container/name-container";
configure({adapter: new Adapter()})

describe('NameContainer', () => {
    const component = shallow(<NameContainer sport="mockSport" eventName="mockEventName"/>);

    it('should render correctly', () => {
        expect(shallowToJson(component)).toMatchSnapshot();
    });
});

describe('NameContainer containers', () => {
    const component = shallow(<NameContainer sport="mockSport" eventName="mockEventName"/>);

    it('should have an image tag with src', () => {
        const container = component.find('.icon-container > img');
        expect(container.props()).toHaveProperty('src', '');
    });

    it('should have a div with class div.game-name', () => {
        expect(component.find('div.game-name').length).toEqual(1)
    })

    it('should have a div with class iv.icon-container', () => {
        expect(component.find('div.icon-container').length).toEqual(1)
    })

    it('should have a div with class div.name-container', () => {
        expect(component.find('div.name-container').length).toEqual(1)
    })
});

