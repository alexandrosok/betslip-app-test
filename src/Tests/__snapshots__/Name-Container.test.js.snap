// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NameContainer should render correctly 1`] = `
<div
  className="name-container"
>
  <div
    className="icon-container"
  >
    <img
      alt=""
      src=""
    />
  </div>
  <div
    className="game-name"
  />
</div>
`;
